ogre/src/OgreLidarVisual.cc

```diff
@@ -0,0 +1,332 @@
+           << std::endl;
+    return;
+  }
+
+  // Process each point from message
+  // Every line segment, and every triangle is saved separately,
+  // as a pointer to a DynamicLine
+  for (unsigned int j = 0; j < this->verticalCount; ++j)
+  {
+    horizontalAngle = this->minHorizontalAngle;
+
+    if (j+1 > this->dataPtr->rayLines.size())
+    {
+      // Ray Strips fill in between the line areas that intersect an object
+      std::shared_ptr<OgreDynamicLines> line =
+                  std::shared_ptr<OgreDynamicLines>(
+                              new OgreDynamicLines(MT_TRIANGLE_STRIP));
+
+      line->setMaterial("Lidar/BlueStrips");
+      std::shared_ptr<Ogre::MovableObject> mv =
+                std::dynamic_pointer_cast<Ogre::MovableObject>(line);
+
+      this->Node()->attachObject(mv.get());
+      this->dataPtr->rayStrips.push_back(line);
+
+      line = std::shared_ptr<OgreDynamicLines>(
+                  new OgreDynamicLines(MT_TRIANGLE_STRIP));
+
+      line->setMaterial("Lidar/LightBlueStrips");
+      mv = std::dynamic_pointer_cast<Ogre::MovableObject>(line);
+      this->Node()->attachObject(mv.get());
+      this->dataPtr->noHitRayStrips.push_back(line);
+
+      line = std::shared_ptr<OgreDynamicLines>(
+                  new OgreDynamicLines(MT_TRIANGLE_FAN));
+
+      line->setMaterial("Lidar/TransBlack");
+      mv = std::dynamic_pointer_cast<Ogre::MovableObject>(line);
+      this->Node()->attachObject(mv.get());
+      this->dataPtr->deadZoneRayFans.push_back(line);
+      this->dataPtr->deadZoneRayFans[j]->AddPoint(
+                  ignition::math::Vector3d::Zero);
+
+
+      line = std::shared_ptr<OgreDynamicLines>(
+                  new OgreDynamicLines(MT_LINE_LIST));
+
+      line->setMaterial("Lidar/BlueRay");
+      mv = std::dynamic_pointer_cast<Ogre::MovableObject>(line);
+      this->Node()->attachObject(mv.get());
+      this->dataPtr->rayLines.push_back(line);
+
+      this->SetVisibilityFlags(0x0FFFFFFF);
+    }
+    this->dataPtr->deadZoneRayFans[j]->SetPoint(0, this->offset.Pos());
+
+
+    unsigned count = this->horizontalCount;
+    // Process each ray in current scan
+    for (unsigned int i = 0; i < count; ++i)
+    {
+      // calculate range of the ray
+      double r = this->dataPtr->laserMsg[ j * this->horizontalCount + i];
+
+      if (verticalAngle - this->maxVerticalAngle > 1.0e-08)
+      {
+        ignerr <<
+            "Vertical angle exceeds maximum limits."
+          << std::endl;
+      }
+
+      if (this->minVerticalAngle - verticalAngle > 1.0e-08 )
+      {
+        ignerr <<
+            "Vertical angle less than minimum limits."
+          << std::endl;
+      }
+
+      if (this->minHorizontalAngle - horizontalAngle > 1.0e-08 )
+      {
+        ignerr <<
+            "Horizontal angle less than minimum limits."
+          << std::endl;
+      }
+
+      if (horizontalAngle - this->maxHorizontalAngle > 1.0e-08)
```

Comment:
would we ever get into a situation that we fail these `if` checks? 

Like mentioned before, we should make sure all the user specified params are validated early on so we don't run into these errors on update.

Reply:
These will not fail with the current conditions, I'll remove them.